kie-server-parent/kie-server-services/kie-server-services-jbpm-ui/src/main/java/org/kie/server/services/jbpm/ui/form/render/AbstractFormRenderer.java

```diff
@@ -449,9 +440,14 @@ protected void processFormLayout(FormInstance topLevelForm,
                                     item.setValue((value != null) ? value.toString() : "");
                                     break;
                             }
-
-                            item.setReadOnly(field.isReadOnly());
-                            item.setRequired(field.isRequired());
+                            Set<String> tags = field.getTags();
+                            if(tags != null) {
```

Comment:
does this mean that tags override any config on the form definition??

Reply:
Hi @pefernan , yes, the tags may have required and read-only property from process data for a field.

Reply:
My only concern here would be that if a field is marked as required or readonly on the form and the field has an empty set of tags, it wil always be set to non-required/readonly. I'm not saying that's wrong, just pointing a possible issue.

Are the tags also taken into account on during the form generation process? Because that would be another interesting point.